tests/import_test.go

```diff
@@ -223,3 +224,63 @@ func startPrometheusPortForward(f *framework.Framework) (string, *exec.Cmd, erro
 
 	return url, cmd, nil
 }
+
+var _ = Describe("Importer Test Suite-Block_device", func() {
+	f := framework.NewFrameworkOrDie(namespacePrefix)
+	var pv *v1.PersistentVolume
+	var storageClass *k8sv1.StorageClass
+	var pod *v1.Pod
+	var err error
+
+	BeforeEach(func() {
+		pod, err = utils.FindPodByPrefix(f.K8sClient, "cdi", "cdi-block-device", "kubevirt.io=cdi-block-device")
+		Expect(err).NotTo(HaveOccurred(), fmt.Sprintf("Unable to get pod %q", "cdi"+"/"+"cdi-block-device"))
+
+		nodeName := pod.Spec.NodeName
+
+		By(fmt.Sprintf("Creating storageClass for Block PV"))
+		storageClass, err = f.CreateStorageClassFromDefinition(utils.NewStorageClassForBlockPVDefinition("manual"))
+		Expect(err).ToNot(HaveOccurred())
+
+		By(fmt.Sprintf("Creating Block PV"))
+		pv, err = f.CreatePVFromDefinition(utils.NewBlockPVDefinition("local-volume", "1G", nil, "manual", nodeName))
+		Expect(err).ToNot(HaveOccurred())
+
+		By("Verify that PV's phase is Available")
+		err = f.WaitTimeoutForPVReady(pv.Name, 60*time.Second)
+		Expect(err).ToNot(HaveOccurred())
+	})
+
+	AfterEach(func() {
+		err := utils.DeletePV(f.K8sClient, pv)
+		Expect(err).ToNot(HaveOccurred())
+
+		err = utils.DeleteStorageClass(f.K8sClient, storageClass)
+		Expect(err).ToNot(HaveOccurred())
+	})
+
+	It("Should create import pod for block pv", func() {
+		httpEp := fmt.Sprintf("http://%s:%d", utils.FileHostName+"."+utils.FileHostNs, utils.HTTPNoAuthPort)
+		pvcAnn := map[string]string{
+			controller.AnnEndpoint: httpEp + "/tinyCore.iso",
+		}
+
+		By(fmt.Sprintf("Creating PVC with endpoint annotation %q", httpEp+"/tinyCore.iso"))
+
+		pvc, err := f.CreatePVCFromDefinition(utils.NewBlockPVCDefinition(
+			"import-image-to-block-pvc",
+			"1G",
+			pvcAnn,
+			nil,
+			"manual"))
+		Expect(err).ToNot(HaveOccurred())
+
+		By("Verify the pod status is succeeded on the target PVC")
+		Eventually(func() string {
+			status, phaseAnnotation, err := utils.WaitForPVCAnnotation(f.K8sClient, f.Namespace.Name, pvc, controller.AnnPodPhase)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(phaseAnnotation).To(BeTrue())
+			return status
+		}, CompletionTimeout, assertionPollInterval).Should(BeEquivalentTo(v1.PodSucceeded))
+	})
```

Comment:
Missing verification of the contents of the block device against the source material (tinycore.iso)

Reply:
same

Reply:
So how are you proving that this is actually working? That is the purpose of the functional tests, to demonstrate the new functionality does what it is supposed to do. transport_test.go has some good examples of us creating a verification pod to determine that the contents we just wrote to the target are actually what we are expecting it to be. Without at least one spot verifying this, we cannot be certain that what is written is actually what it should be.

I must insist on verification of the written data here.